### PR TITLE
defaults: enable 'smarttab' by default.

### PR DESCRIPTION
**Pull request by fmoralesc**
Re: https://github.com/neovim/neovim/issues/2676

Changed legacy test 038 because it assumed 'nosmarttab', and it didn't seem to be meaningful to the result of the test.

### CODE REVIEW

File: test/functional/legacy/038_virtual_replace_spec.lua [hunk 1]
```diff
@@ -10,6 +10,8 @@ describe('Virtual replace mode', function()
   it('is working', function()
     -- Make sure that backspace works, no matter what termcap is used.
     execute('set t_kD=x7f t_kb=x08')
+    -- Use vi default for 'smarttab'
+    execute('set nosmarttab')
```
Comment (justinmk) [inline test/functional/legacy/038_virtual_replace_spec.lua]: do you mean "Vim default"?

Comment (fmoralesc) [inline test/functional/legacy/038_virtual_replace_spec.lua]: Well, technically, vim's default here is to use vi's default, but I can change it if its clearer the other way.

**Comment by justinmk** [comment]
LGTM. Merged

**Comment by fmoralesc** [comment]
Thanks.

**Pull request closed without merging**